code/datums/martial.dm

```diff
@@ -66,12 +66,11 @@
 		H.verbs += help_verb
 	if(make_temporary)
 		temporary = 1
-	if(H.martial_art && H.martial_art.temporary)
-		if(temporary)
+	if(temporary)
+		if(H.martial_art)
```

Comment:
This should also check `&& !H.martial_art.temporary`, you don't want to set the base to something temporary, as you can end up with things like permanent boxing if you mix this with drunken brawling. The one downside is that if you equip boxing gloves, then get really drunk, you'll lose both martial arts once the drunkenness wears off unless you un-equip and re-equip the gloves, but that's preferable to permanent boxing.